auth-service/src/main/java/com/auth/authservice/security/CustomBasicAuthenticationEntryPoint.java

```diff
@@ -0,0 +1,27 @@
+package com.auth.authservice.security;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.authentication.www.BasicAuthenticationEntryPoint;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+
+@Component
+public class CustomBasicAuthenticationEntryPoint extends BasicAuthenticationEntryPoint {
```

Comment:
What does this class do?

Reply:
It catches error thrown in the filter, and sends the message. Normally this is used to redirect when a logout is done. But since the errors are weird, they are caught by this class.